### PR TITLE
feature: categories

### LINKED ISSUE #12: Create product category CRUD
Acceptance Criteria:

1. Implement create product category endpoint POST /category
2. Implement get all user endpoint GET /category
3. Implement get user endpoint GET /category/{id}
4. Implement update user endpoint PATCH /category/{id}
- Only update the field based on the provided request
5. Implement delete user endpoint DELETE /category/{id}

### PR DESCRIPTION
**Pull request by ezabintangr**
### Description
Added features that allow users to create categories for their products. The feature requires user id and category name as a condition to create categories. This feature has also come with get, update, and delete categories.

### Related Issue
fixes #12 

### Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### Additional Information
In this PR, I fixed a bug in the delete user feature

### CODE REVIEW

File: features/user/service/service.go [hunk 1]
```diff
@@ -95,7 +95,12 @@ func Delete(id string) error {
 		return errors.New("you must login first")
 	}
 
-	err := repository.DeleteAccount(id)
+	result, err := GetProfile(id)
+	if err != nil {
+		return err
+	}
+
+	err = repository.DeleteAccount(result.ID)
```
Comment (heroahmadi) [inline features/user/service/service.go:103]: why is this changed?

Comment (ezabintangr) [reply features/user/service/service.go:103]: because when I tried to fill the id in the endpoint with a random string (id that is not in the database), the status became 204. not an error. So I fixed it

**Review by heroahmadi — APPROVED** [review]
